### PR TITLE
XSS Sanitizer

### PR DESCRIPTION
**Pull request by aneeskhan47**

### CODE REVIEW

**Review by ArondeParon — COMMENTED** [review]
Thanks for the pull request! It's always nice to see someone contributing.

I have some questions and doubts regarding this sanitizer though:

- Usually `htmlspecialchars` is used to sanitize data when it is being displayed. This is what Blade does underwater when you render a variable with curly braces, for example. (IE: `{{ $variable }}` will basically do `echo htmlspecialchars($variable);`. I am not sure whether it is a good idea to run it when processing storing data in the database.
- The second argument of `htmlspecialchars` is a flag that can be represented by a constant (`ENT_COMPAT`, `ENT_QUOTES`, etc). Passing the constant value (the number `0`) makes it a bit hard to see what is going on.
- The `TrimTest` seems to be incorrectly named.

In short: I see what you are trying to do here, but I think this use case is not generic enough to be included in the package.

**Comment by aneeskhan47** [comment]
Hmm so what would you suggest?

**Comment by ArondeParon** [comment]
If you need this sanitizer in your project, I would suggest that you implement it outside of the package.
